llama-cpp.nix

```diff
@@ -0,0 +1,118 @@
+{ lib
+, stdenv
+, cmake
+, ninja
+, pkg-config
+, symlinkJoin
+, python3
```

Comment:
When packaging a `buildPythonApplication`, it's preferable to take `python3Packages` as argument (you can access the interpreter as `python3Packages.python`) because of splicing

Reply:
I think I'm going to leave this one as-is from the previous flake.nix, since I don't _really_ understand what better looks like. We can always continue to iterate on it.

Reply:
Ah sorry, missed your comment. Sure, we can change that later. I linked the documentation in the other message though